pkg/sql/plan/shuffle.go

```diff
@@ -339,23 +339,28 @@ func determinShuffleForJoin(n *plan.Node, builder *QueryBuilder) {
 	}
 
 	// get the column of left child
-	var expr *plan.Expr
+	var expr0, expr1 *plan.Expr
 	cond := n.OnList[idx]
 	switch condImpl := cond.Expr.(type) {
 	case *plan.Expr_F:
-		expr = condImpl.F.Args[0]
```

Comment:
What happens to function with 3 args?

Reply:
> What happens to function with 3 args?

The join condition will never have 3 args, it's like "left.col = right.col" or "Function(left.col) < Function(right.col)"